### PR TITLE
[bitnami/elasticsearch] add persistentVolumeClaimRetentionPolicy to statefulset for master and data

### LINKED ISSUE #27846: [bitnami/elasticsearch] Support persistentVolumeClaimRetentionPolicy
### Name and Version

elasticsearch:21.2.8

### What is the problem this feature will solve?

Would enable the ability to clean up volumes automatically as Elasticsearch scales up and down (statefulset) 

### What is the feature you are proposing to solve the problem?

https://github.com/giuliocalzolari/charts/blob/ef4e7d40f0dfd22d6b3672a3514eb3a21c1c27e8/bitnami/rabbitmq/values.yaml#L970C1-L970C37 - persistentVolumeClaimRetentionPolicy - this would solve it and is enabled in several other bitnami charts (https://github.com/search?q=org%3Abitnami+persistentVolumeClaimRetentionPolicy&type=code)

This should probably just be commonly enabled across all Bitnami charts, to be honest... 

### What alternatives have you considered?

None - persistentVolumeClaimRetentionPolicy should be the solution here as it's a K8s native solution - https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-retention

### PR DESCRIPTION
**Pull request by mike12806**
### Description of the change

This change adds support for PVC retention for statefulsets in Elasticsearch data and master https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-retention

### Benefits

Automatic cleanup (if user desires) of PVCs when a statefulset auto scales in 

### Possible drawbacks

None - by default this will be off and user won't lose their PVC on scale in 

### Applicable issues

- fixes #27846

### Additional information

This is already enabled across several other charts - https://github.com/search?q=org%3Abitnami+persistentVolumeClaimRetentionPolicy&type=code

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by mike12806** [comment]
@javsalgar  not sure why the readme check is failing - can you let me know? It seems fine to me...

File: bitnami/elasticsearch/Chart.yaml [hunk 1]
```diff
@@ -34,4 +34,4 @@ maintainers:
 name: elasticsearch
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/elasticsearch
-version: 21.2.8
+version: 21.3.8
```
Comment (migruiz4) [inline bitnami/elasticsearch/Chart.yaml:37]: Patch number should be reset to 0 on minor versions:

```suggestion
version: 21.3.0
```

File: bitnami/elasticsearch/templates/master/statefulset.yaml [hunk 2]
```diff
@@ -391,6 +391,11 @@ spec:
           persistentVolumeClaim:
             claimName: {{ .Values.master.persistence.existingClaim }}
   {{- else }}
+  {{- if .Values.persistentVolumeClaimRetentionPolicy.enabled }}
+          persistentVolumeClaimRetentionPolicy:
+            whenDeleted: {{ .Values.persistentVolumeClaimRetentionPolicy.whenDeleted }}
+            whenScaled: {{ .Values.persistentVolumeClaimRetentionPolicy.whenScaled }}
+  {{- end }}
```
Comment (migruiz4) [inline bitnami/elasticsearch/templates/master/statefulset.yaml:398]: According to [Kubernetes documentation](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-retention) this settings are supposed to be located under `spec.persistentVolumeClaimRetentionPolicy`.

In this case, indent is wrong and these settings should be aligned with `volumeClaimTemplates`, not with `volumes`.

Comment (migruiz4) [inline bitnami/elasticsearch/templates/master/statefulset.yaml:398]: You forgot the `master/data` in the values, it should be `.Values.<component>`

```yaml
  {{- if .Values.master.persistentVolumeClaimRetentionPolicy.enabled }}
          persistentVolumeClaimRetentionPolicy:
            whenDeleted: {{ .Values.master.persistentVolumeClaimRetentionPolicy.whenDeleted }}
            whenScaled: {{ .Values.master.persistentVolumeClaimRetentionPolicy.whenScaled }}
  {{- end }}
```

File: bitnami/elasticsearch/values.yaml [hunk 3]
```diff
@@ -832,6 +832,21 @@ master:
...
+    ##
+    enabled: false
+    ## @param master.persistentVolumeClaimRetentionPolicy.whenScaled Volume retention behavior when the replica count of the StatefulSet is reduced
+    ##
+    whenScaled: Retain
+    ## @param master.persistentVolumeClaimRetentionPolicy.whenDeleted Volume retention behavior that applies when the StatefulSet is deleted
+    ##
+    whenDeleted: Retain
+
+
```
Comment (migruiz4) [inline bitnami/elasticsearch/values.yaml:849]: Please remove extra white lines

**Comment by mike12806** [comment]
> Hi @mike12806,
> 
> Thank you for your contribution! Could you please take a look at my comments?

Should be ready for re-review - thank you for the feedback!

File: bitnami/elasticsearch/templates/master/statefulset.yaml (hunk 2 above)
Comment (mike12806) [reply bitnami/elasticsearch/templates/master/statefulset.yaml]: `helm install elasticsearch bitnami/elasticsearch --dry-run` is working now - thanks!

**Review by migruiz4 — APPROVED** [review]
Thank you for your contribution @mike12806!

I submitted a commit fixing a typo in the values.yaml, renaming 'Primary Statefulset' as 'Master/Data Statefulset' respectively.

I tested the changes manually and everything works as expected, so it looks good to me!

**Pull request merged**